tasks/libs/common/omnibus.py

```diff
@@ -16,190 +16,115 @@
-            "GOPROXY",
-            "HELM_HOOKS_CI_IMAGE",
-            "HELM_HOOKS_PERIODICAL_REBUILD_CONDUCTOR_ENV",
-            "HOME",
-            "HOSTNAME",
-            "HOST_IP",
-            "INFOPATH",
-            "INSTALL_SCRIPT_API_KEY_ORG2",
-            "INSTANCE_TYPE",
-            "INTEGRATION_WHEELS_CACHE_BUCKET",
-            "IRBRC",
-            "KITCHEN_INFRASTRUCTURE_FLAKES_RETRY",
-            "LANG",
-            "LESSCLOSE",
-            "LESSOPEN",
-            "LC_CTYPE",
-            "LS_COLORS",
-            "MACOS_S3_BUCKET",
-            "MANPATH",
-            "MESSAGE",
-            "NEW_CLUSTER",
-            "NEW_CLUSTER_PR_SLACK_WORKFLOW_WEBHOOK",
-            "NOTIFICATIONS_SLACK_CHANNEL",
-            "NOTIFIER_IMAGE",
-            "OLDPWD",
-            "PCP_DIR",
-            "PACKAGE_ARCH",
-            "PIP_EXTRA_INDEX_URL",
-            "PIP_INDEX_URL",
-            "PIPELINE_KEY_ALIAS",
-            "PROCESS_S3_BUCKET",
-            "PWD",
-            "PROMPT",
-            "RESTORE_CACHE_ATTEMPTS",
-            "RUSTC_SHA256",
-            "SIGN",
-            "SHELL",
-            "SHLVL",
-            "SLACK_AGENT",
-            "STATIC_BINARIES_DIR",
-            "STATSD_URL",
-            "SYSTEM_PROBE_BINARIES_DIR",
-            "TEAM_ID",
-            "TIMEOUT",
-            "TMPDIR",
-            "TRACE_AGENT_URL",
-            "USER",
-            "USERDOMAIN",
-            "USERNAME",
-            "USERPROFILE",
-            "VCPKG_BLOB_SAS_URL",
-            "VERSION",
-            "VIRTUAL_ENV",
-            "VM_ASSETS",
-            "WIN_S3_BUCKET",
-            "WINGET_PAT",
-            "WORKFLOW",
-            "_",
-            "_OLD_VIRTUAL_PS1",
-            "__CF_USER_TEXT_ENCODING",
-            "build_before",
-        ]
-        for p in excluded_prefixes:
-            if key.startswith(p):
-                return False
-        for s in excluded_suffixes:
-            if key.endswith(s):
-                return False
-        if key in excluded_values:
-            return False
-        return True
-
-    return dict(filter(env_filter, sorted(os.environ.items())))
+    excluded_variables = {
+        'APPDATA',
+        'DEB_GPG_KEY',
+        'DEB_GPG_KEY_NAME',
+        'DEB_SIGNING_PASSPHRASE',
+        'GEM_HOME',
+        'GEM_PATH',
+        'HOME',
+        'JARSIGN_JAR',
+        'LOCALAPPDATA',
+        'MY_RUBY_HOME',
+        'OMNIBUS_GIT_CACHE_DIR',
+        'OMNIBUS_GOMODCACHE',
+        'OMNIBUS_WORKERS_OVERRIDE',
+        'PACKAGE_VERSION',
+        'PYTHONUTF8',
+        'RPM_GPG_KEY',
+        'RPM_GPG_KEY_NAME',
+        'RPM_SIGNING_PASSPHRASE',
+        'S3_OMNIBUS_CACHE_ANONYMOUS_ACCESS',
+        'SIGN_WINDOWS_DD_WCS',
+        'SYSTEMDRIVE',
+        'SYSTEMROOT',
+        'SSL_CERT_FILE',
+        'TEMP',
+        'TMP',
+        'USERPROFILE',
+        'rvm_bin_path',
+        'rvm_path',
+        'rvm_prefix',
+        'rvm_version',
+    }
```

Comment:
in the above list, whereas excluding temporary directory & cache management variables makes completely sense, I'm more concerned by variables that are -at least semantically- subject to alter the generated artifacts, for instance:
- signing related: wouldn't different keys/certificates lead to different signatures?
- compilation toolchain related: wouldn't a different `rvm` lead to a different `omnibus-ruby` behavior?

Reply:
I don't think it will matter much in practice (those variables seldom change), but I'm also not going to die on this hill. 
My thinking for these was:
- Signing happens always out of (after) everything that we currently consider cacheable is cached.
- rvm version would only affect ruby behavior if something's very broken with rvm. rvm is a "version manager" for ruby, meaning it's used as a way to get the version of ruby that we want; we should be getting the exact same version of ruby regardless of rvm version.